sdk/couchbase-core/src/memdx/response.rs

```diff
@@ -1211,6 +1213,405 @@ impl TryFromClientResponse for DecrementResponse {
+                    return Err(Error::protocol_error("bad value length"));
+                }
+                let mut cursor = Cursor::new(value);
+
+                let op_index = cursor
+                    .read_u8()
+                    .map_err(|e| Error::from(ErrorKind::Protocol { msg: e.to_string() }))?;
+                let res_status = cursor
+                    .read_u16::<BigEndian>()
+                    .map_err(|e| Error::from(ErrorKind::Protocol { msg: e.to_string() }))?;
+                let res_status = Status::from(res_status);
+
+                let err_kind: SubdocErrorKind = match res_status {
+                    Status::SubDocDocTooDeep => SubdocErrorKind::DocTooDeep,
+                    Status::SubDocNotJSON => SubdocErrorKind::NotJSON,
+                    Status::SubDocPathNotFound => SubdocErrorKind::PathNotFound,
+                    Status::SubDocPathMismatch => SubdocErrorKind::PathMismatch,
+                    Status::SubDocPathInvalid => SubdocErrorKind::PathInvalid,
+                    Status::SubDocPathTooBig => SubdocErrorKind::PathTooBig,
+                    Status::SubDocPathExists => SubdocErrorKind::PathExists,
+                    Status::SubDocCantInsert => SubdocErrorKind::CantInsert,
+                    Status::SubDocBadRange => SubdocErrorKind::BadRange,
+                    Status::SubDocBadDelta => SubdocErrorKind::BadDelta,
+                    Status::SubDocValueTooDeep => SubdocErrorKind::ValueTooDeep,
+                    _ => SubdocErrorKind::UnknownStatus {status: res_status},
+                };
+
+                return Err(ServerError::new(
+                    ServerErrorKind::Subdoc{ error: SubdocError::new(err_kind, Some(op_index)) },
+                    resp.packet(),
+                    resp.local_addr(),
+                    resp.peer_addr()
+                ).into());
+            }
+        }
+
+        let mut doc_is_deleted = false;
+        if status == Status::SubDocSuccessDeleted {
+            doc_is_deleted = true;
+            // still considered a success
+        } else if status != Status::Success && status != Status::SubDocMultiPathFailure {
+            return Err(OpsCrud::decode_common_error(
+                resp.packet(),
+                resp.local_addr(),
+                resp.peer_addr(),
+            ));
+        }
+
+        let mut results: Vec<SubDocResult> = Vec::with_capacity(subdoc_info.op_count as usize);
+
+        if let Some(value) = &resp.packet().value {
+            let mut cursor = Cursor::new(value);
+
+            while cursor.position() < cursor.get_ref().len() as u64 {
+                if cursor.remaining() < 3 {
+                    return Err(Error::protocol_error("bad value length"));
+                }
+
+                let op_index = cursor
+                    .read_u8()
+                    .map_err(|e| Error::from(ErrorKind::Protocol { msg: e.to_string() }))?;
+
+                if op_index > results.len() as u8 {
```

Comment:
How can the op_index be greater than results.len()?

Reply:
Say that we received a value on Operation 0 and 2, but not on Operation 1, during the second iteration: `op_index == 2`, and `results.len() == 1`, so we need to fill all the results in-between with empty SubDocResults